apps/studio/src/features/editing-experience/components/ScheduledPublishingModal/ScheduledPublishingModal.tsx

```diff
@@ -0,0 +1,427 @@
+import { trpc } from "~/utils/trpc"
+
+interface ScheduledPublishingModalProps extends UseDisclosureReturn {
+  pageId: number
+  siteId: number
+  // callback invoked to handle the publishing process now
+  onPublishNow: (pageId: number, siteId: number) => void
+  // used for the loading state of the publish button
+  isPublishingNow?: boolean
+}
+
+/**
+ * The times to display as 'quick-select-times' under the date/time picker
+ */
+const QUICK_SELECT_TIMES: { hours: number; minutes: number }[] = [
+  { hours: 0, minutes: 0 },
+  { hours: 9, minutes: 0 },
+  { hours: 13, minutes: 0 },
+  { hours: 17, minutes: 0 },
+]
+
+export const ScheduledPublishingModal = ({
+  pageId,
+  siteId,
+  onClose,
+  onPublishNow,
+  isPublishingNow,
+  ...rest
+}: ScheduledPublishingModalProps): JSX.Element => {
+  const toast = useToast()
+  const utils = trpc.useUtils()
+  const [isScheduledPublishValid, setIsScheduledPublishValid] = useState(false)
+
+  const methods = useZodForm<typeof publishClientSchema>({
+    schema: publishClientSchema,
+    defaultValues: {
+      publishMode: PublishMode.NOW,
+      pageId,
+      siteId,
+    },
+  })
+  const publishMode = methods.watch("publishMode")
+  // Validate form when publish mode changes or when the date/time selected changes
+  // so the banner can be shown to the users. Do NOT use trigger() since that
+  // triggers validation and causes error messages to appear
+  useEffect(() => {
+    const validateForm = () => {
+      if (publishMode === PublishMode.NOW) setIsScheduledPublishValid(false)
+      const valid = publishClientSchema.safeParse(methods.getValues())
+      setIsScheduledPublishValid(valid.success)
+    }
+    const subscription = methods.watch(() => void validateForm())
+    return () => subscription.unsubscribe()
+  }, [methods, publishMode])
+
+  const { mutate: schedulePageMutation, isPending: isScheduling } =
+    trpc.page.schedulePage.useMutation({
+      onSettled: async () => {
+        await utils.page.readPage.refetch({ pageId, siteId })
+        onClose()
+      },
+      onSuccess: () => {
+        toast({
+          status: "success",
+          title: "Page scheduled successfully",
+          ...BRIEF_TOAST_SETTINGS,
+        })
+      },
+      onError: (error) => {
+        console.error(`Error occurred when scheduling page: ${error.message}`)
+        toast({
+          status: "error",
+          title: "Failed to schedule page. Please contact Isomer support.",
+          ...BRIEF_TOAST_SETTINGS,
+        })
+      },
+    })
+  return (
+    <Modal onClose={onClose} {...rest}>
+      <form
+        onSubmit={methods.handleSubmit(
+          (res: z.output<typeof publishClientSchema>) => {
+            // publish immediately if publish mode is now, else schedule publish
+            if (res.publishMode === PublishMode.NOW) {
+              onPublishNow(res.pageId, res.siteId)
+            } else {
+              schedulePageMutation(res)
+            }
```

Comment:
nit - while we currenty only have `PublishMode.NOW` and `PublishMode.SCHEDULED`, perhaps we can use a switch statement here with exhaustic check, so that if/when we add another method, it will throw type error